packages/legend-extension-dsl-serializer/src/components/studio/SchemaSetElementEditor.tsx

```diff
@@ -0,0 +1,285 @@
+        <DropdownMenu
+          disabled={isReadOnly}
+          content={
+            <MenuContent className="schema-set-format-editor__dropdown">
+              {Object.values(FORMAT_TYPE).map((format) => (
+                <MenuContentItem
+                  key={format}
+                  className="schema-set-format-editor__option"
+                  onClick={changeType(format)}
+                >
+                  {format}
+                </MenuContentItem>
+              ))}
+            </MenuContent>
+          }
+          menuProps={{
+            anchorOrigin: { vertical: 'bottom', horizontal: 'right' },
+            transformOrigin: { vertical: 'top', horizontal: 'right' },
+          }}
+        >
+          <div className="schema-set-format-editor__type">
+            <div className="schema-set-format-editor__type__label">
+              {schemaSet.format}
+            </div>
+            <div className="schema-set-format-editor__type__icon">
+              <CaretDownIcon />
+            </div>
+          </div>
+        </DropdownMenu>
+      </div>
+    );
+  },
+);
+
+const SchemaBasicEditor = observer(
+  (props: { schema: Schema; isReadOnly: boolean }) => {
+    const { schema, isReadOnly } = props;
+    const changeId: React.ChangeEventHandler<HTMLInputElement> = (event) =>
+      schema.setId(event.target.value);
+    const changeLocation: React.ChangeEventHandler<HTMLInputElement> = (
+      event,
+    ) => schema.setLocation(event.target.value);
+    const editorOptions: editor.IEditorOptions & editor.IGlobalEditorOptions = {
+      lineNumbers: 'off',
+      fontFamily: 'Arial',
+      lineDecorationsWidth: 0,
+    };
+    return (
+      <div className="schema-editor">
+        <input
+          className="schema-editor__id"
+          disabled={isReadOnly}
+          value={schema.id}
+          spellCheck={false}
+          onChange={changeId}
+          placeholder={`Id`}
+        />
+        <input
+          className="schema-editor__location"
+          disabled={isReadOnly}
+          value={schema.location}
+          spellCheck={false}
+          onChange={changeLocation}
+          placeholder={`Location`}
+        />
+        <div className={clsx('schema-editor__content')}>
+          <div className="schema-editor__content__input">
+            <TextInputEditor
+              inputValue={schema.content}
+              language={EDITOR_LANGUAGE.TEXT}
+              updateInput={(val: string): void => {
+                schema.setContent(val);
+              }}
+              extraEditorOptions={editorOptions}
+            />
+          </div>
+        </div>
+      </div>
+    );
+  },
+);
+
+export const SchemaSetEditor = observer(() => {
+  const editorStore = useEditorStore();
+  const editorState = editorStore.getCurrentEditorState(SchemaSetEditorState);
+  const schemaSet = editorState.schemaSet;
+  const isReadOnly = editorState.isReadOnly;
+  schemaSet.setFormat(FORMAT_TYPE.FLAT_DATA);
```

Comment:
This is a `no-no`, you're modifying the state in the render method.

Reply:
Instead, when you initialize the state, you could have set this if this is not set already, if you do this, it means whatever the format is, it's always reset to `FLAT_DATA` which I believe is not what we intend to do